src/hangar/dataloaders/torch_loader.py

```diff
@@ -0,0 +1,97 @@
+from ..dataset import DatasetDataReader
+try:
+    import torch
+    from torch.utils.data import Dataset, DataLoader
+except (ImportError, ModuleNotFoundError):
+    raise RuntimeError("Could not import torch. Try installing extra dependencies")
+
+
+class TorchDataSet(Dataset):
+    """
+    TorchDataSet inherits `torch.utils.data.Dataset` and override
+    `__len__` & `__getitem__`. Not needed to use directly by the user.
+
+    Parameters
+    ----------
+    hangar_datasets : `hangar.dataset.DatasetDataReader` or a Collection
+        A dataset object or a list of dataset objects which will be
+        consumed by :class:`TorchDataSet`
+    """
+
+    def __init__(self, hangar_datasets):
+        if isinstance(hangar_datasets, DatasetDataReader):
+            hangar_datasets = [hangar_datasets]
+
+        try:
+            self._dataset_len = len(hangar_datasets[0])
+        except IndexError:
+            raise
+        except Exception:
+            raise TypeError("`hangar_datasets` has to be a valid hangar dataset or an"
+                            "iterable that can hold other objects, like a list.")
+
+        for dset in hangar_datasets:
+            if not isinstance(dset, DatasetDataReader):
+                raise TypeError("`hangar_dataset` contains invalid hangar dataset(s)")
+            if len(dset) != self._dataset_len:
+                raise RuntimeError("Got datasets with different length")
+        self._number_of_datasets = len(hangar_datasets)
+        self.hangar_datasets = hangar_datasets
+        self._data_names = tuple(self.hangar_datasets[0].keys())
+
+    def __len__(self):
+        return self._dataset_len
+
+    def __getitem__(self, index):
+        key = self._data_names[index]
+        if self._number_of_datasets == 1:
+            return torch.from_numpy(self.hangar_datasets[0][key])
+        out = []
+        for dset in self.hangar_datasets:
+            out.append(torch.from_numpy(dset[key]))
+        return out
+
+
+class TorchLoader(DataLoader):
```

Comment:
I would remove this class unless there are plans to add functionality to it. As it stands, there is no advantage to the user for using it.

Reply:
I disagree. It lets the user import only one loader and there is no need for doing the whole Dataset creation call. For those who need more control, we can make the `hangar.dataloaders.TorchDataSet` available. I agree with the fact that it doesn't do a lot but I see it as a convenience. After all, it doesn't harm.
That being said, I don't have any sentiment for keeping that, so if everyone thinks that should be gone, I am OK with that too.

Reply:
Okay, @elistevens I am going to apologize for the disagreement. With the introduction of new keys, I couldn't find a way to keep the TorchLoader clean which is worthy enough. So removing it.